pkg/apiserver/event/event.go

```diff
@@ -47,11 +51,45 @@ func NewService(
 
 // Register mounts our HTTP handler on the mux.
 func Register(r *gin.RouterGroup, s *Service) {
-	endpoint := r.Group("/event")
+	endpoint := r.Group("/events")
 
 	// TODO: add more api handlers
 	endpoint.GET("/all", s.listEvents)
 }
 
-// TODO: need to be implemented
-func (s *Service) listEvents(c *gin.Context) {}
+func (s *Service) listEvents(c *gin.Context) {
+	name := c.Query("name")
+	namespace := c.Query("namespace")
+	eventList := make([]*core.Event, 0)
+
+	if name == "" && namespace == "" {
+		eventList, err := s.event.List(context.Background())
+		if err != nil {
+			c.JSON(http.StatusOK, gin.H{
+				"status": statuscode.GetResourcesFromDBWrong,
+				"message": "get events wrong",
+				"data": eventList,
+			})
+			return
+		}
+	} else if (name != "" && namespace == "") || (name == "" && namespace != "") {
```

Comment:
So we must have both `name` and `namespace`? I think there is a case that I only want to see the events in one namespace. WDYT?

Reply:
Understand, I will modify it.